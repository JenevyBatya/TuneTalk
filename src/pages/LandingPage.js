import React from 'react';
import Slider from 'react-slick';
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";
import photo1 from "../assets/rec1.jpg";
import photo2 from "../assets/rec2.jpg";
import photo3 from "../assets/rec3.jpg";
import auth1 from "../assets/auth1.jpg"
import auth2 from "../assets/auth2.jpg"
import auth3 from "../assets/auth3.jpg"
import {
    HeadingText1,
    HeadingText2,
    MainContainer,
    RecBlock,
    RegBlock,
    StyledButton,
    Text2,
    Text1,
    HeaderBlock,
    PhotoDiv,
    Photo, RecommendDiv
} from "../styles/LandingPageStyles";
import {Link} from "react-router-dom";
import Bugsnag from "@bugsnag/js";
import HeaderComponent from '../components/HeaderComponent';
import FooterNavigation from '../components/FooterComponent';

const images = [photo1, photo2, photo3];
const authors = [auth1, auth2, auth3];
const LandingPage = () => {
    const settings = {
        dots: true,
        infinite: true,
        speed: 500,
        slidesToShow: 1,
        slidesToScroll: 1,
        autoplay: true,
        autoplaySpeed: 3000
    };
    Bugsnag.notify(new Error('Test error'))

    return (
        <MainContainer>
            <div>
            {/*<HeaderBlock></HeaderBlock>*/}
                <HeaderComponent/>
            <RecBlock>
                <HeadingText1>
                    Хостинг подкастов
                </HeadingText1>
                <Text1>
                    Все русскоязычные подкасты на одной платформе. Возможность переключаться между аудио и видео
                    форматами в один клик. Удобный поиск. Эксклюзивные подборки.
                </Text1>
                <PhotoDiv>
                    <Slider {...settings}>
                        {images.map((src, index) => (
                            <Photo key={index} src={src} alt={`photo ${index + 1}`}
                                   style={{width: '100%', height: 'auto'}}/>
                        ))}
                    </Slider>


                </PhotoDiv>
                <RecommendDiv>
                    Рекомендуем посмотреть
                </RecommendDiv>
            </RecBlock>
            <RegBlock>
                <HeadingText2>
                    Любимые авторы в одном месте
                </HeadingText2>
                <Text2>
                    Мы собрали всех ваших любимых авторов, чтобы вы наслаждались ими без лишних сложностей.
                </Text2>
                <PhotoDiv>
                    <Slider {...settings}>
                        {authors.map((src, index) => (
                            <Photo
                                key={index}
                                src={src}
                                alt={`auth ${index + 1}`}
                                style={{
                                    width: '100%',
                                    height: 'auto',
                                    objectFit: 'cover'
                                }}
                            />
                        ))}
                    </Slider>

                </PhotoDiv>
                <RecommendDiv>
                <StyledButton as={Link} to="/Registration">
                    Зарегистрироваться
                </StyledButton>
                </RecommendDiv>

            </RegBlock>
            </div>

            <FooterNavigation/>
        </MainContainer>
    );
}

export default LandingPage;
