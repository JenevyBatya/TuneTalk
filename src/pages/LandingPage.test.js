import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import LandingPage from './LandingPage';
import Slider from 'react-slick';
import { axe, toHaveNoViolations } from 'jest-axe';


jest.mock('../components/FooterComponent', () => () => <div>Footer</div>);
jest.mock('../components/HeaderComponent', () => () => <div>Header</div>);
jest.mock('react-slick', () => ({ children }) => <div>{children}</div>);

expect.extend(toHaveNoViolations);

describe('LandingPage Component', () => {
    it('renders the header component', () => {
        render(
            <Router>
                <LandingPage />
            </Router>
        );
        expect(screen.getByText('Header')).toBeInTheDocument();
    });

    it('renders the registration button with correct link', () => {
        render(
            <Router>
                <LandingPage />
            </Router>
        );
        const registerButton = screen.getByRole('link', { name: 'Зарегистрироваться' });
        expect(registerButton).toHaveAttribute('href', '/Registration');
    });

    it('renders the slider with images', () => {
        render(
            <Router>
                <LandingPage />
            </Router>
        );
        const images = screen.getAllByRole('img');
        expect(images).toHaveLength(6);
        images.forEach((image, index) => {
            expect(image).toHaveAttribute('src');
            expect(image).toHaveAttribute('alt', expect.stringMatching(/photo|auth/));
        });
    });

    it('renders the footer component', () => {
        render(
            <Router>
                <LandingPage />
            </Router>
        );
        expect(screen.getByText('Footer')).toBeInTheDocument();
    });

    it('matches the snapshot', () => {
        const { asFragment } = render(
            <Router>
                <LandingPage />
            </Router>
        );
        expect(asFragment()).toMatchSnapshot();
    });


});
