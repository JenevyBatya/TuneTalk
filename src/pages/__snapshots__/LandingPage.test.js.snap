// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb dWkdZC"
  >
    <div>
      <div>
        Header
      </div>
      <div
        class="sc-egkSDF Jeplm"
      >
        <div
          class="sc-dntaoT garjUI"
        >
          Хостинг подкастов
        </div>
        <div
          class="sc-ivxoEo inRjlb"
        >
          Все русскоязычные подкасты на одной платформе. Возможность переключаться между аудио и видео форматами в один клик. Удобный поиск. Эксклюзивные подборки.
        </div>
        <div
          class="sc-kLhKbu bdlupj"
        >
          <div>
            <img
              alt="photo 1"
              class="sc-ixGGxD feERQe"
              src="rec1.jpg"
              style="width: 100%; height: auto;"
            />
            <img
              alt="photo 2"
              class="sc-ixGGxD feERQe"
              src="rec2.jpg"
              style="width: 100%; height: auto;"
            />
            <img
              alt="photo 3"
              class="sc-ixGGxD feERQe"
              src="rec3.jpg"
              style="width: 100%; height: auto;"
            />
          </div>
        </div>
        <div
          class="sc-dpBQxM iIhucL"
        >
          Рекомендуем посмотреть
        </div>
      </div>
      <div
        class="sc-dstKZu jXwizn"
      >
        <div
          class="sc-fAUdSK kLKyXb"
        >
          Любимые авторы в одном месте
        </div>
        <div
          class="sc-ghWlax fgRdex"
        >
          Мы собрали всех ваших любимых авторов, чтобы вы наслаждались ими без лишних сложностей.
        </div>
        <div
          class="sc-kLhKbu bdlupj"
        >
          <div>
            <img
              alt="auth 1"
              class="sc-ixGGxD feERQe"
              src="auth1.jpg"
              style="width: 100%; height: auto; object-fit: cover;"
            />
            <img
              alt="auth 2"
              class="sc-ixGGxD feERQe"
              src="auth2.jpg"
              style="width: 100%; height: auto; object-fit: cover;"
            />
            <img
              alt="auth 3"
              class="sc-ixGGxD feERQe"
              src="auth3.jpg"
              style="width: 100%; height: auto; object-fit: cover;"
            />
          </div>
        </div>
        <div
          class="sc-dpBQxM iIhucL"
        >
          <a
            class="sc-khLCKb bcnzYB"
            href="/Registration"
          >
            Зарегистрироваться
          </a>
        </div>
      </div>
    </div>
    <div>
      Footer
    </div>
  </div>
</DocumentFragment>
`;
